stubs/redis/redis/client.pyi

```diff
@@ -684,38 +684,297 @@ class PubSub:
+    def sentinel_master(self, service_name) -> Pipeline: ...
+    def sentinel_masters(self) -> Pipeline: ...
+    def sentinel_monitor(self, name, ip, port, quorum) -> Pipeline: ...
+    def sentinel_remove(self, name) -> Pipeline: ...
+    def sentinel_sentinels(self, service_name) -> Pipeline: ...
+    def sentinel_set(self, name, option, value) -> Pipeline: ...
+    def sentinel_slaves(self, service_name) -> Pipeline: ...
+    def shutdown(self) -> None: ...
+    def slaveof(self, host=..., port=...) -> Pipeline: ...
+    def slowlog_get(self, num=...) -> Pipeline: ...
+    def slowlog_len(self) -> Pipeline: ...
+    def slowlog_reset(self) -> Pipeline: ...
+    def time(self) -> Pipeline: ...
+    def append(self, key, value) -> Pipeline: ...
+    def bitcount(self, key: _Key, start: Optional[int] = ..., end: Optional[int] = ...) -> Pipeline: ...
+    def bitop(self, operation, dest, *keys) -> Pipeline: ...
+    def bitpos(self, key, bit, start=..., end=...) -> Pipeline: ...
+    def decr(self, name, amount=...) -> Pipeline: ...
+    def delete(self, *names: _Key) -> Pipeline: ...
+    def __delitem__(self, _Key) -> None: ...
+    def dump(self, name) -> Pipeline: ...
+    def exists(self, *names: _Key) -> Pipeline: ...
+    def __contains__(self, *names: _Key) -> Pipeline: ...
+    def expire(self, name: _Key, time: Union[int, timedelta]) -> Pipeline: ...
+    def expireat(self, name, when) -> Pipeline: ...
+    def get(self, name: _Key) -> Pipeline: ...
+    def __getitem__(self, name) -> Pipeline: ...
+    def getbit(self, name: _Key, offset: int) -> Pipeline: ...
+    def getrange(self, key, start, end) -> Pipeline: ...
+    def getset(self, name, value) -> Pipeline: ...
+    def incr(self, name, amount=...) -> Pipeline: ...
+    def incrby(self, name, amount=...) -> Pipeline: ...
+    def incrbyfloat(self, name, amount=...) -> Pipeline: ...
+    def keys(self, pattern=...) -> Pipeline: ...
+    def mget(self, keys, *args) -> Pipeline: ...
+    def mset(self, *args, **kwargs) -> Pipeline: ...
+    def msetnx(self, *args, **kwargs) -> Pipeline: ...
+    def move(self, name, db) -> Pipeline: ...
+    def persist(self, name) -> Pipeline: ...
+    def pexpire(self, name, time) -> Pipeline: ...
+    def pexpireat(self, name, when) -> Pipeline: ...
+    def psetex(self, name, time_ms, value) -> Pipeline: ...
+    def pttl(self, name) -> Pipeline: ...
+    def randomkey(self) -> Pipeline: ...
+    def rename(self, src, dst) -> Pipeline: ...
+    def renamenx(self, src, dst) -> Pipeline: ...
+    def restore(self, name, ttl, value) -> Pipeline: ...
+    def set(
+        self,
+        name: _Key,
+        value: _Value,
+        ex: Union[None, int, timedelta] = ...,
+        px: Union[None, int, timedelta] = ...,
+        nx: bool = ...,
+        xx: bool = ...,
+        keepttl: bool = ...,
+    ) -> Pipeline: ...
+    def __setitem__(self, name, value) -> None: ...
```

Comment:
should all the dunder methods be organized at the top?

Reply:
Personally, I try to use the order the implementation uses. But it is your choice.